zebedee-cms/src/main/java/com/github/onsdigital/zebedee/api/Publish.java

```diff
@@ -67,6 +87,74 @@ public boolean publish(HttpServletRequest request, HttpServletResponse response)
         }
     }
 
+    /**
+     * Manual Publish the files in a collection
+     *
+     * @param collection       the collection to publish
+     * @param session          a session with editor priviledges
+     * @param skipVerification
+     * @return success
+     * @throws IOException
+     * @throws UnauthorizedException
+     * @throws BadRequestException
+     * @throws ConflictException     - If there
+     */
+    public boolean publish(Collection collection,
+                           Session session,
+                           boolean breakBeforePublish,
+                           boolean skipVerification)
+            throws IOException, UnauthorizedException, BadRequestException,
+            ConflictException, NotFoundException {
+
+        // Collection exists
+        if (collection == null) {
+            throw new BadRequestException("Please provide a valid collection.");
+        }
+
+        // User has permission
+        if (session == null || !zebedeeCmsService.getPermissions().canEdit(session.getEmail())) {
+            throw new UnauthorizedException(getUnauthorizedMessage(session));
+        }
+
+        // Check approval status
+        if (collection.description.approvalStatus != ApprovalStatus.COMPLETE) {
+            throw new ConflictException("This collection cannot be published because it is not approved");
+        }
+
+        // Break before transfer allows us to run tests on the prepublish-hook without messing up the content
+        if (breakBeforePublish) {
+            logInfo("Breaking before publish").log();
+            return true;
+        }
+        logInfo("Going ahead with publish").log();
+
+        Keyring keyring = zebedeeCmsService.getZebedee().getKeyringCache().get(session);
+        if (keyring == null) throw new UnauthorizedException("No keyring is available for " + session.getEmail());
```

Comment:
Can you add a log message before throwing the execption

Reply:
Done